react-frontend/src/profile/ChangeProfilePicture.js

```diff
@@ -8,59 +8,80 @@ import DefaultPfp from '../assets/profile-placeholder.png';
 function ChangeProfilePicture(props) {
   const [profileImage, setProfileImage] = useState('');
   const [showNotification, setShowNotification] = useState(false);
+  const [isSuccessful, setIsSuccessful] = useState(false);
   const [notificationMsg, setNotificationMsg] = useState('');
   const [loading, setLoading] = useState(false);
+  const [timestamp, setTimestamp] = useState(Date.now());
+  
 
   useEffect(() => {
-    // Fetch isProfilePicture value when the component mounts
     fetchIsProfilePicture();
   }, []);
 
   const fetchIsProfilePicture = async () => {
     try {
       const response = await axios.get(`${process.env.REACT_APP_BACKEND_LINK}/users/is-profile-picture/${props.userID}`);
       const { isProfilePicture } = response.data;
-      // If isProfilePicture is true, set the profile image to the uploaded image
-      if (isProfilePicture) {
-        setProfileImage(`https://haggleimgs.s3.amazonaws.com/user/${props.userID}/bruh0.jpg`);
-      } else {
-        // If isProfilePicture is false, set the profile image to the default placeholder image
-        setProfileImage(DefaultPfp);
-      }
+      setProfileImage(isProfilePicture ? `https://haggleimgs.s3.amazonaws.com/user/${props.userID}/bruh0.jpg?${timestamp}` : DefaultPfp);
     } catch (error) {
       console.error('Error fetching isProfilePicture:', error);
     }
   };
 
   const handleFileChange = async (event) => {
+    const MINIMUM_IMAGE_WIDTH = 100;
+    const MINIMUM_IMAGE_HEIGHT = 100;
     const file = event.target.files[0];
     const userID = props.userID;
-    if (!file) return;
-    console.log(file);
+
+    if (!file || event.target.files.length > 1) {
+      displayNotification("Please submit one image");
+      return;
+    }
+
     try {
       setLoading(true);
       const formData = new FormData();
-      formData.append('file', file);
       formData.append('userID', userID);
-      console.log(userID);
 
-      console.log('FormData:', formData);
+      const validateImage = (file) => {
+        return new Promise((resolve, reject) => {
+          const reader = new FileReader();
+          reader.onload = (e) => {
+            const img = new Image();
+            img.src = e.target.result;
+            img.onload = () => {
+              if (img.width < MINIMUM_IMAGE_WIDTH || img.height < MINIMUM_IMAGE_HEIGHT) {
```

Comment:
I think we should also have a maximum resolution so that our storage doesn't take a hit